umbraco/Umbraco-CMS
# V10: Reintroduce appsettings-schema.json

This builds upon #12413 

With V10 we swapped to using RCL for all the static content, however, this meant that we lost the `appsettings-schema.json` file, this PR re-introduces this, by copying it from `Umbraco.Cms` on build when you're using the templates. It also moves the file to live in the root now.

The schema will be generated in the `Umbraco.Cms` project and the `Umbraco.Web.UI` project if it does not exist.

When the schema is copied it's done with `SkipUnchangedFiles` enabled, so it'll only copy if there are any changes, this mean that if you install a new version you will automatically get the new intellisense

## Breaking change
This is a breaking change since it moves the location of the `appsettings-schema.json` file, this means when upgrading you have to open your appsettings files and change the schema definition to point to the new file like so: 

Change 
```json
"$schema" : "./umbraco/config/appsettings-schema.json",
```

To
```json
"$schema" : "./appsettings-schema.json",
```

## Testing 

1. Create a new site using the templates
2. Ensure that `appsettings-schema.json` is placed in the root when building the project for the first time 
3. Ensure that the appsettings files have intellisense


### Addendum 

The easiest way to do the testing is to bump the version in the `Directory.Build.props` file and the `template.json` files, and then pack the solution by doing `dotnet pack umbraco.sln -o TestBuild`, then copy all the nuget files to a local feed and use that for testing.
